From: =?utf-8?q?Thomas_B=C3=A4chler?= <thomas@archlinux.org>
Date: Sun, 15 Dec 2013 12:06:37 +0100
Subject: login: Don't stop a running user manager from garbage-collecting the
 user.

With the current logic, a user will never be garbage-collected, since its
manager will always be around. Change the logic such that a user is
garbage-collected when it has no sessions and linger is disabled.

Conflicts:
	src/login/logind-user.c
---
 src/login/logind-user.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/login/logind-user.c b/src/login/logind-user.c
index adbe638..e2fce5c 100644
--- a/src/login/logind-user.c
+++ b/src/login/logind-user.c
@@ -629,12 +629,6 @@ int user_check_gc(User *u, bool drop_not_started) {
         if (u->slice_job || u->service_job)
                 return 1;
 
-        if (u->slice && manager_unit_is_active(u->manager, u->slice) != 0)
-                return 1;
-
-        if (u->service && manager_unit_is_active(u->manager, u->service) != 0)
-                return 1;
-
         return 0;
 }
 
